tools/depends/target/ffmpeg/Makefile

```diff
@@ -38,6 +38,7 @@ ifeq ($(OS), android)
       ffmpg_config += --cpu=cortex-a9
     else
       ffmpg_config += --cpu=i686 --disable-mmx
+      ffmpg_config += --extra-cflags=-no-integrated-as --extra-cflags=-mno-stackrealign
```

Comment:
The idea behind https://github.com/xbmc/xbmc/pull/14615/files#diff-c1f8ad28b2cf0ee2006db6bdcf0656afR24 was to move the extra options to configure.ac to not have multiple places where we do our own magic.

Reply:
My suggestion was a general one, not OS dependent. For new devs its quite hard to find such hidden treasures. We usualy handle all platform / os stuff in our configure, no?

But I'm Ok with everything, Its nothing I maintain.

So, from my POV all fine :-)

Reply:
But what really wonders me is why we sat i686, didn't we have aarch64 which is != arm ?

Reply:
I'm happy to move the change to another location, I tried to follow recent commits, I believe there was a tweak for mips that was done in the Makefile for ffmpeg.  Please just give me a hint where; I'm not very familiar with the Kodi build process outside of Windows.

For what it's worth, the Kodi APK that was generated does appear to work in an x86 emulator.  There is a text relocation warning message on Android that would need to be resolved, but otherwise it fires right up and works.  I would be happy to try and correct the text relocation message and submit a follow-up PR.  I would also be happy to re-execute a full build with the binary addons and make sure they all compile.

I don't think subsequent actions for x86 would necessarily make sense in the context of this PR, though.  Again, just let me know!

Reply:
@djp952 : After https://github.com/xbmc/xbmc/blob/master/tools/depends/configure.ac#L225 would be the place where to set, line 254 in this file shows how to set.

Reply:
The configure file is the wrong place, as there is no possible variant to compile for android x86 without this flags.
Keep it as is.

Reply:
?? And having it in ffmpeg's Makefile gives you more variants? Can you explain pls?

Reply:
Makefile is for non optional configure options like this one, as you can't compile for android x86 without them.
The `--with-ffmpeg-options` parameter in configure is to let users enable or disable optional features (like vaapi, vdpau , ...). But setting this option overdries the default ones specified in the file and then you would have to pass the parameters (which aren't optional) too (after first getting a build failure).

Reply:
Thx for explanation, makes sense